### PR TITLE
feat: added UNet example to experimental [DET-3397]

### PR DESCRIPTION
**Pull request by naren-determined**
## Description
UNet example on tf.keras (https://www.tensorflow.org/tutorials/images/segmentation).
TODO: YogaDL Data Porting

## Test Plan
Tested functionality in both single GPU and distributed training contexts on p2.8xlarge instance on AWS.

## Commentary (optional)
For when we get to playing around with our own datasets:
http://www.aylward.org/notes/open-access-medical-image-repositories (See Cancer Imaging Archive)
https://github.com/beamandrew/medical-data (See Oasis)
https://github.com/beamandrew/medical-data

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: examples/experimental/trial/unets_tf_keras/const.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+description: unets_tf_keras_const
```
Comment (katport) [inline examples/experimental/trial/unets_tf_keras/const.yaml:1]: don't forget a readme

File: examples/experimental/trial/unets_tf_keras/const.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  name: single
+  metric: val_accuracy
+  smaller_is_better: false
+  max_steps: 230 # There are ~7400 examples (200 examples/class * 37 classes) in the training set
+                   # We run for 20 epochs as per the Tensorflow UNet tutorial. We get 230 by:
+                   # 7400 / global_batch_size / (1 batch per step) * 20 epochs
+
+min_validation_period: 10
+entrypoint: model_def:UNetsTrial
+max_restarts: 0
```
Comment (katport) [inline examples/experimental/trial/unets_tf_keras/const.yaml:24]: remove

File: examples/experimental/trial/unets_tf_keras/model_def.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
+from typing import List
```
Comment (katport) [inline examples/experimental/trial/unets_tf_keras/model_def.py:1]: class comments

File: examples/experimental/trial/unets_tf_keras/const.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+description: unets_tf_keras_const
+data:
+  train_url: http://download.tensorflow.org/data/iris_training.csv 
+  test_url: http://download.tensorflow.org/data/iris_test.csv
```
Comment (seanr15) [inline examples/experimental/trial/unets_tf_keras/const.yaml:4]: Are these necessary?

Comment (naren-determined) [reply examples/experimental/trial/unets_tf_keras/const.yaml:4]: @katport this is for if we choose to use our own datasets, right? If the plan is to merge this into master as a trial example before the City of Hope demo, should we keep this or not?

Comment (katport) [reply examples/experimental/trial/unets_tf_keras/const.yaml:4]: these should be removed - it's leftover from the iris example. 

@naren-determined we try to minimize the amount of extra code to examples. When we add new data we can redesign the const.yaml.

**Pull request closed without merging**